### DISCUSSION TITLE
Can we configure tunnel config/swanctl.conf in such a way that it will accept certificate if it is coming from IP address which is mentioned in certificate DNS field?

### DISCUSSION THREAD

**Discussion by digambaringale333** [strongswan/strongswan#811] [Q&A]
Hi,
We are working on application where we are using strongswan to create secure tunnel between two endpoints. We are using pubkey based authentication, let's say we have generated a certificate with some IP address/DNS name.

Is there any way in swanctl.conf OR strongswan.conf which enable strongswan(local end point) to accepts the certificate only if it is coming from IP address(remote end point) which is mentioned in the DNS filed of certificate ?

**Comment by tobiasbrunner** [comment]
No, subjectAltName extensions of type `dNSName` are never resolved to an IP address. And SANs of type `iPAddress` are also never matched against a peer's IP address but its identity (IDi/IDr) if it's of type `ID_IPV4_ADDR` or `ID_IPV6_ADDR`. And to enforce that the identity matches the IP address, it would have to be configured explicitly as expected remote identity.
